tests/granular/test_nounphase.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+import numpy as np
+
+from nlp_profiler.noun_phase_count import count_noun_phase, gather_nouns
+from nlp_profiler.constants import NaN
+
+
+sentence = "European authorities fined Google a record $5.1 billion on Wednesday \
+            for abusing its power in the mobile phone market and ordered the company \
+            to alter its practices"
+
+text_to_return_value_mapping = [
+    (sentence, [('European', 'NORP'), ('Google', 'ORG'), ('$5.1 billion', 'MONEY'), ('Wednesday', 'DATE')])
+]
+
+@pytest.mark.parametrize("text,expected_result",
+                         text_to_return_value_mapping)
+def test_gather_noun(text: str, expected_result: str):
+    
+    actual_result = gather_nouns(text)
+    
+    assert actual_result == expected_result, \
+    f"Expected Result: {expected_result} \
+    Actual Result: {actual_result}"
+
+
+text_to_return_value_mapping = [
+    ("", [])
+]
+
+@pytest.mark.parametrize("text,expected_result",
+                         text_to_return_value_mapping)
+def test_failed_gather_noun(text: str, expected_result: str):
```

Comment:
I don't think this test is necessary, it could be combined with the `test_gather_noun` because there is no such thing as failure, unless there is an exception. 

It either is a sentence or not a sentence. The "is not a sentence" like sentences are edge-cases which may need to be gathered and put there to improve robustness of the test.

It's possible the nltk library handles non-sentences well.

Reply:
Okay I'll remove it. I added to make a seperation between correct and false cases.